wavelength-plugins/news.js

```diff
@@ -82,25 +99,52 @@ exports.commands = {
 			this.modlog(`NEWS`, null, `Added announcement: ${parts[0]}`);
 			this.privateModAction(`(${user.name} added server announcement: ${parts[0]})`);
 		},
+
 		subscribe: function (target, room, user) {
 			if (!user.named) return this.errorReply('You must choose a name before subscribing');
 			if (Db.NewsSubscribers.has(user.userid)) return this.errorReply("You are alreading subscribing Wavelength News.");
 			Db.NewsSubscribers.set(user.userid, true);
 			this.sendReply("You have subscribed Wavelength News.");
 			this.popupReply("|wide||html|You will receive Wavelength News automatically once you connect to the Wavelength next time.<br><hr><center><button class='button' name='send' value ='/news'>View News</button></center>");
 		},
+
 		unsubscribe: function (target, room, user) {
 			if (!user.named) return this.errorReply('You must choose a name before unsubscribing');
 			if (!Db.NewsSubscribers.has(user.userid)) return this.errorReply("You have not subscribed Wavelength News.");
 			Db.NewsSubscribers.remove(user.userid);
 			this.sendReply("You have unsubscribed Wavelength News.");
 			this.popupReply("|wide||html|You will no longer automatically receive Wavelength News.<br><hr><center><button class='button' name='send' value='/news'>View News</button></center>");
 		},
+
+		request: function (target, room, user) {
+			if (!user.named) return this.errorReply("You must have a name before requesting an announcement.");
+			if (!this.canTalk()) return this.errorReply("You can't use this command while unable to speak.");
+			if (!target) return this.sendReply(`/news request [message] - Requests a news announcement from the Wavelength Staff.`);
+			if (target.length < 1) return this.sendReply(`/news request [message] - Requests a news announcement from the Wavelength Staff.`);
+			let newsId = (Object.keys(newsRequests).length + 1);
+			let d = new Date();
+			let MonthNames = ["January", "February", "March", "April", "May", "June",
+				"July", "August", "September", "October", "November", "December",
+			];
+			while (newsRequests[newsId]) newsId--;
```

Comment:
Why does this exist?

Reply:
I'm not even sure when this would actually be evaluated. Isn't `newsId` always going to be the latest index, which should (theoretically) never collide with the index before it?